Extension.php

```diff
@@ -43,6 +43,14 @@ public function initialize()
             }
         }
 
+        if (!empty($lang) && $this->isValidLanguage($lang)) {
+            $this->app['session']->set('lang', $lang);
+        }
+        if (is_null($this->app['session']->get('lang'))) {
+            $this->app['session']->set('lang', 'nl');
+        }
+        $this->setCurrentLanguage($lang);
```

Comment:
Can we please move session handling to a before() midldeware function as this will have impact for performance an caching.

Also what language should be the fallback?